Conch/lib/Conch/Model/DeviceService.pm

```diff
@@ -0,0 +1,230 @@
+=head1 NAME
+
+Conch::Model::DeviceService
+
+=head1 DESCRIPTION
+
+A service represents functionality for a device. 
+
+=cut
+
+package Conch::Model::DeviceService;
+
+use strict;
+use warnings;
+use utf8;
+use v5.20;
+
+use Moo;
+use Moo::Role::ToJSON;
+use experimental qw(signatures);
+
+use Try::Tiny;
+use Type::Tiny;
+use Types::Standard qw(Str);
+use Types::UUID qw(Uuid);
+
+use Conch::Pg;
+
+with("Moo::Role::ToJSON");
+with("Conch::Role::But");
+with("Conch::Role::Timestamps");
+
+
+
+=head1 ACCESSORS
+
+=over 4
+
+=item id
+
+UUID. Cannot be written by user.
+
+=cut
+
+has 'id' => (
+	is  => 'rwp',
+	isa => Uuid,
+);
+
+
+=item name
+
+String. Required.
+
+=cut
+
+has 'name' => (
+	is       => 'rw',
+	isa      => Str,
+	required => 1,
+);
+
+
+sub _build_serializable_attributes {[qw[
+	id
+	name
+	created
+	updated
+	deactivated
+]]}
+
+
+
+=back 4
+
+=head1 METHODS
+
+=head2 from_id
+
+	my $service = Conch::Model::DeviceService->from_id($uuid);
+
+Returns a DeviceService given an existing UUID.
+
+Includes deactivated objects
+
+=cut
+
+sub from_id ($class, $uuid) {
+	$class->_from(id => $uuid);
+}
+
+
+
+=head2 from_name
+
+	my $service = Conch::Model::DeviceService->from_name($name);
+
+Returns a DeviceService given an existing name
+
+Includes deactivated objects
```

Comment:
Same issue as mentioned above. Deactivated and recreating with the same name may only return the deactivated one.